modules/algebrasAndModel/src/main/scala/utils/Types.scala

```diff
@@ -68,13 +78,121 @@ object Types {
+    implicit val decoder: Decoder[TeamDomain] = deriveUnwrappedDecoder[TeamDomain]
+  }
+
+  final case class EnterpriseId(enterpriseId: String) extends AnyVal
+  object EnterpriseId {
+    implicit val encoder: Encoder[EnterpriseId] = deriveUnwrappedEncoder[EnterpriseId]
+    implicit val decoder: Decoder[EnterpriseId] = deriveUnwrappedDecoder[EnterpriseId]
+  }
+
+  final case class EnterpriseName(enterpriseName: String) extends AnyVal
+  object EnterpriseName {
+    implicit val encoder: Encoder[EnterpriseName] =
+      deriveUnwrappedEncoder[EnterpriseName]
+    implicit val decoder: Decoder[EnterpriseName] =
+      deriveUnwrappedDecoder[EnterpriseName]
+  }
+
+  final case class ChannelId(channelId: String) extends AnyVal
+  object ChannelId {
+    implicit val encoder: Encoder[ChannelId] = deriveUnwrappedEncoder[ChannelId]
+    implicit val decoder: Decoder[ChannelId] = deriveUnwrappedDecoder[ChannelId]
+  }
+
+  final case class ChannelName(channelName: String) extends AnyVal
+  object ChannelName {
+    implicit val encoder: Encoder[ChannelName] = deriveUnwrappedEncoder[ChannelName]
+    implicit val decoder: Decoder[ChannelName] = deriveUnwrappedDecoder[ChannelName]
+  }
+  final case class UserId(userId: String) extends AnyVal
+  object UserId {
+    implicit val encoder: Encoder[UserId] = deriveUnwrappedEncoder[UserId]
+    implicit val decoder: Decoder[UserId] = deriveUnwrappedDecoder[UserId]
+  }
+
+  final case class UserName(userName: String) extends AnyVal
+  object UserName {
+    implicit val encoder: Encoder[UserName] = deriveUnwrappedEncoder[UserName]
+    implicit val decoder: Decoder[UserName] = deriveUnwrappedDecoder[UserName]
+  }
+
+  final case class Command(command: String) extends AnyVal
+  object Command {
+    implicit val encoder: Encoder[Command] = deriveUnwrappedEncoder[Command]
+    implicit val decoder: Decoder[Command] = deriveUnwrappedDecoder[Command]
+  }
+
+  final case class Text(text: String) extends AnyVal
+  object Text {
+    implicit val encoder: Encoder[Text] = deriveUnwrappedEncoder[Text]
+    implicit val decoder: Decoder[Text] = deriveUnwrappedDecoder[Text]
+  }
+
+  final case class ResponseUrl(responseUrl: String) extends AnyVal
+  object ResponseUrl {
+    implicit val encoder: Encoder[ResponseUrl] = deriveUnwrappedEncoder[ResponseUrl]
+    implicit val decoder: Decoder[ResponseUrl] = deriveUnwrappedDecoder[ResponseUrl]
+  }
+
+  final case class TriggerId(triggerId: String) extends AnyVal
+  object TriggerId {
+    implicit val encoder: Encoder[TriggerId] = deriveUnwrappedEncoder[TriggerId]
+    implicit val decoder: Decoder[TriggerId] = deriveUnwrappedDecoder[TriggerId]
+  }
+
+  final case class ApiAppId(apiAppId: String) extends AnyVal
+  object ApiAppId {
+    implicit val encoder: Encoder[ApiAppId] = deriveUnwrappedEncoder[ApiAppId]
+    implicit val decoder: Decoder[ApiAppId] = deriveUnwrappedDecoder[ApiAppId]
+  }
```

Comment:
We should consider using `newtype` in the following PR and remove all these `deriveUnwrappedEncoder`s. We can see it together.